### PR TITLE
Codenet subsets rework

### PR DESCRIPTION
**Pull request by kmh4321**
This PR adds the 2 subsets of the CodeNet dataset. Corresponding changes are made for notebooks and references.

### CODE REVIEW

**Comment by kmh4321** [comment]
cc: @ckadner 
I signed off the commits every time but the CI check still fails.

File: dataset-samples/codenet_langclass/codenet_langclass.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+seo_tags:
+  - Solutions to Programming Problems
+  - Text
+
+# OPTIONAL; assets that complement this data set, e.g. notebooks
+related_assets:
+  - name: Project CodeNet Language Classification
+    description: Simple experiment that shows how to create and exercise a Keras model to detect the language of a piece of source code
+    mime_type: application/x-ipynb+json
+    url: https://github.com/machine-learning-exchange/katalog/blob/main/notebook-samples/src/dlf-notebooks/Project_CodeNet_LangClass.ipynb
```
Comment (ckadner) [inline dataset-samples/codenet_langclass/codenet_langclass.yaml:56]: the `src` should not be under `dlf-notebook` but rather under `https://github.com/machine-learning-exchange/katalog/blob/main/notebook-samples/src/codenet/` both notebooks should fit

File: dataset-samples/codenet_mlm/codenet_mlm.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  name: Data Asset eXchange
+  url: https://developer.ibm.com/exchanges/data/all/project-codenet/
+
+# OPTIONAL; but recommended
+seo_tags:
+  - Solutions to Programming Problems
+  - Text
+
+# OPTIONAL; assets that complement this data set, e.g. notebooks
+related_assets:
```
Comment (ckadner) [inline dataset-samples/codenet_mlm/codenet_mlm.yaml:52]: the `related_assets` also need the application `mlx` part:

```YAML
related_assets:
    ...
    mime_type: application/x-ipynb+json
    url: ...
    application:
        name: MLX
        asset_id: notebooks/project-codenet-language-classification
```

**Review by ckadner — CHANGES_REQUESTED** [review]
Thanks Karthik. Just a few changes than we can merge and iterate over what's needed next

File: dataset-samples/codenet_mlm/codenet_mlm.yaml [hunk 3]
```diff
@@ -53,7 +53,10 @@ related_assets:
   - name: Project CodeNet Masked Language Model
     description: Experiment investigates whether a popular attention model to construct a masked language model (MLM) can be used for source code instead of natural language sentences
     mime_type: application/x-ipynb+json
-    url: https://github.com/machine-learning-exchange/katalog/blob/main/notebook-samples/src/dlf-notebooks/Project_CodeNet_MLM.ipynb
+    url: https://github.com/machine-learning-exchange/katalog/blob/main/notebook-samples/src/codenet/Project_CodeNet_MLM.ipynb
+    application:
+      name: MLX
+      asset_id: notebooks/project-codenet-mlm
```
Comment (ckadner) [inline dataset-samples/codenet_mlm/codenet_mlm.yaml:59]: the asset IDs get generated from the name, so the related asset_id is

```YAML
      asset_id: notebooks/project-codenet-masked-language-model
```

See the URL here: https://ml-exchange.org/notebooks/project-codenet-masked-language-model

File: dataset-samples/codenet_langclass/codenet_langclass.yaml [hunk 4]
```diff
@@ -53,6 +53,9 @@ related_assets:
   - name: Project CodeNet Language Classification
     description: Simple experiment that shows how to create and exercise a Keras model to detect the language of a piece of source code
     mime_type: application/x-ipynb+json
-    url: https://github.com/machine-learning-exchange/katalog/blob/main/notebook-samples/src/dlf-notebooks/Project_CodeNet_LangClass.ipynb
+    url: https://github.com/machine-learning-exchange/katalog/blob/main/notebook-samples/src/codenet/Project_CodeNet_LangClass.ipynb
+    application:
+      name: MLX
+      asset_id: notebooks/project-codenet-language-classification
```
Comment (ckadner) [inline dataset-samples/codenet_langclass/codenet_langclass.yaml:59]: 👍🏻

**Review by ckadner — CHANGES_REQUESTED** [review]
one last change request -- otherwise looks good 👍🏻

**Review by ckadner — APPROVED** [review]
Great -- thanks @kmh4321 !

/lgtm

**Pull request merged**